### DISCUSSION TITLE
ShoopDaLoop v0.1.0 released

### DISCUSSION THREAD

**Discussion by SanderVocke** [SanderVocke/shoopdaloop#427] [Announcements]
Hi there!

Not many people following this repo, but regardless I'll make a small post to announce the first versioned development release of ShoopDaLoop: v0.1.0.

Until a v1.0.0, don't expect this software to be especially stable or to have all the features you would expect. Still, it is a milestone because it took a lot of work to get the tooling and automation running to be able to release often and fast.

Also, I am playing around with this a lot and have been having a lot of fun making small looping sessions.

From now on, from dev release to dev release there will be a changelog to keep track of what changed. Some of the things that have happened in the last few months:

- Improved builds and packaging
- Lots of bugfixes and optimizations
- Adding ARM support (only Mac and Linux so far) - once GitHub makes ARM CI runners public later this year, Windows can also be supported

It really is about time I start making some demo videos...

**Comment by g1l3sp** [comment]
Hi Sander, I've been following your project ever since I stumbled onto the Reddit post where you announced it, and I'm very enthusiastic about what you are making. I was successfully trying it out before the Rust bits came into the mix, but now I'm unable to build it, and I can't get the AppImages from your builds to launch successfully on my debian laptop. What Linux distro and version do you recommend for the smoothest path to getting recent builds to run? Thanks!

**Reply by SanderVocke** [reply]
Hi @g1l3sp, nice to hear you've been following the project!

I reached a point where it was pretty much feature-complete in my eyes, but there are still some bugs and lack of polish. I think migrating to Rust and getting rid of the Python parts will help make it more reliable in the long run. But it's going to be quite a while to finish that process, with some stumbling blocks along the way.

I think the smoothest path is indeed to run the AppImages. But you are right: I also noticed this week that the latest AppImage on main doesn't run anymore. It took a while for me to notice because it still somehow runs correctly in CI.

I would recommend sticking with the pre-Rust version (so essentially: this 1.0 release) for a bit while I look into the AppImage issue on main, then move to using the AppImages from main again. I will make sure to test on Debian (BTW, which version are you on?)

I created an issue for the AppImage being broken: #484 .

Of course, if you would like some help to get through your build issues I would be happy to advise. The gist of it is that you should have Python, a C++ toolchain and the **nightly** Rust toolchain installed, plus the Python dependencies listed in build_python_requirements.txt (`pip install -r build_python_requirements.txt`). Then, running `cargo build` in the top-level directory should do it.

Be aware that hardly anything changed in terms of features since 1.0, so you're not missing much at the moment!

**Reply by SanderVocke** [reply]
Another thing: please let me know your thoughts! I tend to get a bit of tunnel vision because I am pretty much the only user. Always interested to hear what you think about the state of the software, be it through bug/feature request issues or a post in the Github discussion forum.

**Reply by SanderVocke** [reply]
I made a change which, at least for me, fixes the AppImage on main. Could you give it a try? If it doesn't work, can you report what is going wrong? Cheers!

**Reply by g1l3sp** [reply]
I think that laptop is running bullseye but I'll have to double check on that. I'll take a look at the new build later today.

**Reply by g1l3sp** [reply]
Okay, I had a few minutes to try it out. I was wrong by the way, the laptop I'm playing with it on is running bookworm.

I downloaded the AppImage artifact from build #2916 and it is running successfully. I generated a click loop for the sync, added a track and recorded a midi loop. Everything good so far! :fireworks:

**Reply by SanderVocke** [reply]
That's good news! I think there has been a bit of performance improvement on main as well. Happy looping!
